src/TextField/TextField.js

```diff
@@ -105,7 +105,7 @@ const getStyles = (props, context, state) => {
  * @returns True if the string provided is valid, false otherwise.
  */
 function isValid(value) {
-  return value !== '' && value !== undefined && value !== null;
+  return value !== '' && value !== undefined && value !== null && !(Array.isArray(value) && value.length === 0);
```

Comment:
Hmm. TextField is reaching down into the child DropDownMenu to read its value, so I don't see a good way to avoid the empty array check in TextField.